[WIP] Add usage page (#1721)
quarkusio/quarkusio.github.io: usage.md

---
layout: usage

> title: Quarkus Usage Analytics

Reviewer: Shouldn't the title be `Quarkus Build Analytics`? or whatever we use in the main repo doc?

Author: What about: `Quarkus Build Analytics Usage` ?

subtitle: 
permalink: /usage/
---

> Starting from Quarkus 3.2 when you run Quarkus in devmode (`quarkus dev`, `mvn quarkus:dev`, etc.) you get asked if you agree to contribute anonymous build data to the Quarkus community.

Author: devmode -> dev mode

> Starting from Quarkus 3.2 when you run Quarkus the first time in dev mode (`quarkus dev`, `mvn quarkus:dev`, etc.) you get asked if you agree to contribute anonymous build data to the Quarkus community.

Reviewer 2: Starting from Quarkus 3.2, when you run Quarkus for the first time in dev mode (`quarkus dev`, `mvn quarkus:dev`, etc.), you are asked if you agree to contribute anonymous build data to the Quarkus community.

> Starting from Quarkus 3.2 when you run Quarkus the first time in dev mode (`quarkus dev`, `mvn quarkus:dev`, etc.) you get asked if you agree to contribute anonymous build data to the Quarkus community.

Reviewer 2: ```suggestion
Starting from Quarkus 3.2, when you run Quarkus for the first time in dev mode (`quarkus dev`, `mvn quarkus:dev`, etc.), you are asked if you agree to contribute anonymous build data to the Quarkus community.
```

![/assets/images/usage-prompt.png](/assets/images/usage-prompt.png)

> If you answer Yes then, when you perform a quarkus build, anonymized data is sent

Reviewer 2: ```suggestion
If you answer 'Yes', then anonymized data will be sent when you perform a quarkus build,
```

> If you answer Yes then when you perform a quarkus build is performed anonymized data is sent
> to gather usage statistics on how Quarkus is being used and adopted.

Author: ```suggestion
If you answer Yes then, when you perform a quarkus build, anonymized data is sent
to gather usage statistics on how Quarkus is being used and adopted.
```

> to gather usage statistics on how Quarkus is being used and adopted.

Reviewer 2: I think this shouldn't be on a new line. Not a biggy, but makes it harder to review and propose changes.

> This service is provided by Red Hat. More details can be found on the  [usage policy](/usage/policy) page.

Reviewer: ```suggestion
This service is provided by Red Hat. More details can be found in the [usage policy](/usage/policy) page.
```

Reviewer 3: will push fix for this though I thought things written is on the page not in ?

If you answer No, you will not be asked again and no usage data is sent.

This information is invaluable for the Quarkus team and contributors to 

> gauge which operating systems, java version, build systems, extensions and more are 

Reviewer 2: ```suggestion
gauge which operating systems, java versions, build systems, extensions and more are 
```

used. 

> We will share what is learned from these anonymous usage data and plan on integrate things like usage and adoption 

Reviewer 2: ```suggestion
We will share what is learned from this anonymous usage data and plan to integrate aggregated forms, such as usage and adoption, 
```

> We will share what is learned from these anonymous usage data and plan on integrate things like usage and adoption 

Reviewer 2: ```suggestion
We will share what is learned from this anonymous usage data and plan to integrate aggregated forms, such as usage and adoption, 
```

into sites like [extensions.quarkus.io](extensions.quarkus.io) and [code.quarkus.io](code.quarkus.io).

## Enabling and Disabling Telemetry Collection

Build time analytics is not active by default. If you have opted in and would like to disable build time analytics, you can do so in two ways:

Globally, by editing the `com.redhat.devtools.quarkus.localconfig` file in the `.redhat` folder of your user’s home directory. Update the file as follows:

```
{"active":false}
```

Per project, by using the system property`quarkus.analytics.disabled=true`

Example using maven:</p>

```
mvn clean install -Dquarkus.analytics.disabled=true
```

> Similar you can set it to `quarkus.analytics.disabled=true` if you want to enable it for a single run.

Author: ```suggestion
```
I wouldn't mention this.
This behaviour was not discussed. Explicit false should bypass the User's prompt question and send analytics?

Reviewer 3: true - i'll delete it.

Reviewer 2: Also, I presume you mean "disable it for a single run".

Mentioning this incase you decide not to delete it.
